platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/InputValidationTest.java

```diff
@@ -44,15 +48,30 @@
 
     private static final String BAD_MIME = "text/html";
 
-    @Before
-    public void setup() {
-        System.setProperty("bad.files",
-                "crossdomain.xml,clientaccesspolicy.xml,.htaccess,.htpasswd,hosts,passwd,group,resolv.conf,nfs.conf,ftpd.conf,ntp.conf,web.config,robots.txt");
-        System.setProperty("bad.file.extensions",
-                ".exe,.jsp,.html,.js,.php,.phtml,.php3,.php4,.php5,.phps,.shtml,.jhtml,.pl,.py,.cgi,.msi,.com,.scr,.gadget,.application,.pif,.hta,.cpl,.msc,.jar,.kar,.bat,.cmd,.vb,.vbs,.vbe,.jse,.ws,.wsf,.wsc,.wsh,.ps1,.ps1xml,.ps2,.ps2xml,.psc1,.psc2,.msh,.msh1,.msh2,.mshxml,.msh1xml,.msh2xml,.scf,.lnk,.inf,.reg,.dll,.vxd,.cpl,.cfg,.config,.crt,.cert,.pem,.jks,.p12,.p7b,.key,.der,.csr,.jsb,.mhtml,.mht,.xhtml,.xht");
-        System.setProperty("bad.mime.types",
-                "text/html,text/javascript,text/x-javascript,application/x-shellscript,text/scriptlet,application/x-msdownload,application/x-msmetafile");
-    }
+    private static final String IGNORE_FILE = ".DS_Store";
+
+    private static final String IGNORE_FILE_CASE_INSENSITIVE = ".dS_sTorE";
+
+    private static final String IGNORE_FILE_KNOWN_GOOD = "valid_ddms_record.xml";
+
+    @ClassRule
+    public static final ProvideSystemProperty badFilesSystemProperties = new ProvideSystemProperty(
+            "bad.files",
+            "crossdomain.xml,clientaccesspolicy.xml,.htaccess,.htpasswd,hosts,passwd,group,resolv.conf,nfs.conf,ftpd.conf,ntp.conf,web.config,robots.txt,Thumbs.db");
+
+    @ClassRule
+    public static final ProvideSystemProperty badFileExtenstionsSystemProperties =
+            new ProvideSystemProperty("bad.file.extensions",
+                    ".exe,.jsp,.html,.js,.php,.phtml,.php3,.php4,.php5,.phps,.shtml,.jhtml,.pl,.py,.cgi,.msi,.com,.scr,.gadget,.application,.pif,.hta,.cpl,.msc,.jar,.kar,.bat,.cmd,.vb,.vbs,.vbe,.jse,.ws,.wsf,.wsc,.wsh,.ps1,.ps1xml,.ps2,.ps2xml,.psc1,.psc2,.msh,.msh1,.msh2,.mshxml,.msh1xml,.msh2xml,.scf,.lnk,.inf,.reg,.dll,.vxd,.cpl,.cfg,.config,.crt,.cert,.pem,.jks,.p12,.p7b,.key,.der,.csr,.jsb,.mhtml,.mht,.xhtml,.xht");
+
+    @ClassRule
+    public static final ProvideSystemProperty badMimeTypesSystemProperties =
+            new ProvideSystemProperty("bad.mime.types",
+                    "text/html,text/javascript,text/x-javascript,application/x-shellscript,text/scriptlet,application/x-msdownload,application/x-msmetafile");
+
+    @ClassRule
+    public static final ProvideSystemProperty ignoreFilesSystemProperties =
```

Comment:
Yes, for all the `InputValidationTest.java` unit tests. If I understand your question correctly.

I thought this was helpful:
https://stackoverflow.com/questions/41121778/junit-rule-and-classrule